### DISCUSSION TITLE
File Extension Changed to .c When Saving xxx.config File

### DISCUSSION THREAD

**Discussion by Seaweed0112** [zed-industries/zed#20468] [Feature Requests]
When I create a file, use Command+S to save, and enter the filename as myconfig.config, Zed automatically changes it to myconfig.config.c, which is not what I intended. I would like Zed to save the filename exactly as I entered it.

I am using MacOS.

![image](https://github.com/user-attachments/assets/538229d7-d8c7-43f0-983c-f96023984b54)

**Comment by SomeoneToIgnore** [comment]
Could that be another part of https://github.com/zed-industries/zed/issues/16969 I wonder?

> use Command+S to save

If this involves a masOS system save dialogue, this has certain issues described above.

**Reply by Seaweed0112** [reply]
Yes! I also noticed the same behavior!

**Reply by SomeoneToIgnore** [reply]
In that case https://github.com/zed-industries/zed/pull/20195 is what you need: that is supposed to be shipped in Preview already and comes to Stable in 3 days.

The PR also mentions a setting as a workaround.

**Reply by Seaweed0112** [reply]
Awesome! Thank you so much. This issue has been bothering me for a long time, and I suddenly thought of bringing it up here. I’m glad to see that others had already noticed it and provided a solution.
